src/Form/Form.js

```diff
@@ -378,11 +423,25 @@ class Form {
     }
 
     addInput (input) {
+        if (this.inputs.all.has(input)) return this
+
+        // If the form has too many inputs, destroy everything to avoid performance issues
+        if (this.inputs.all.size > MAX_INPUTS_PER_FORM) {
```

Comment:
And here we `destroy()` the form, removing all listeners and icons, if the form grows over a certain threshold.

Reply:
love it!